docker-compose/docker-compose.yaml

```diff
@@ -1,5 +1,53 @@
 version: '2.4'
 services:
+  # Description: This container will migrate the Postgres `frontend`,
+  # `codeintel`, and `codeinsights` databases
+  #
+  # Disk: None
+  # Ports: None
+  #
+  # This container should start after the database instance starts, perform the migrations
+  # and then exit cleanly. After exiting the `frontend` service is allowed to start.
+  #
+  # If you're using a DB instance hosted outside of docker-compose, the environment variables
+  # for this container will need to be updated to reflect the new connection information.
+  migrator:
+    container_name: migrator
+    image: 'index.docker.io/sourcegraph/migrator:122149_2021-12-17_1f7179c'
+    cpus: 4
+    mem_limit: '8g'
+    command:
+      ["up"]
+    environment:
+      - PGHOST=pgsql
+      - PGPORT=5432
+      - PGUSER=sg
+      - PGPASSWORD=sg
+      - PGDATABASE=sg
+      - PGSSLMODE=disable
+
+      - CODEINTEL_PGHOST=codeintel-db
+      - CODEINTEL_PGPORT=5432
+      - CODEINTEL_PGUSER=sg
+      - CODEINTEL_PGPASSWORD=sg
+      - CODEINTEL_PGDATABASE=sg
+      - CODEINTEL_PGSSLMODE=disable
+
+      - CODEINSIGHTS_PGHOST=codeinsights-db
```

Comment:
(Not really relevant to this PR) Are there any plans to update the frontend `CODEINSIGHTS_PGDATASOURCE` connection string to follow this same pattern? It's less relevant for docker deployments but the difference in format makes secure credential management difficult to manage.

Reply:
I'm honestly not sure. I agree, it's a little odd having it be different. `codeinsights` also doesn't use the same `sg`/`sg` pattern that the other DB's use.

We had a brief discussion about it on slack here: https://sourcegraph.slack.com/archives/C02FPB08MN3/p1642189372001800?thread_ts=1642106589.009300&cid=C02FPB08MN3